messaging/writer_test.go

```diff
@@ -26,41 +27,67 @@ func TestNewWriter(t *testing.T) {
 func TestMissyWriter_WriteSuccess(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	brokerWriterMock := NewMockBrokerWriter(mockCtrl)
-	msg := &Message{Key: []byte("key"), Value: []byte("value")}
+	msg := &Message{Key: []byte("key"), Value: []byte("value"), Time: tm.Unix(0, 0).UTC()}
 
 	brokerWriterMock.EXPECT().WriteMessages(gomock.Any(), *msg).Return(nil)
 	brokerWriterMock.EXPECT().Close().Return(nil)
 
 	writer := missyWriter{brokerWriter: brokerWriterMock}
 
+	exec := false
+
+	monkey.Patch(tm.Now, func() tm.Time {
+		exec = true
+		return tm.Unix(0, 0).UTC()
+	})
+
+	defer monkey.Unpatch(tm.Now)
+
 	key := []byte("key")
 	value := []byte("value")
 
 	if err := writer.Write(key, value); err != nil {
 		t.Error("there was an unexpected error during Write message")
 	}
 
+	if !exec {
+		t.Error("function patching was not called!")
+	}
+
 	defer writer.Close()
 
 }
 
 func TestMissyWriter_WriteError(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	brokerWriterMock := NewMockBrokerWriter(mockCtrl)
-	msg := &Message{Key: []byte("key"), Value: []byte("value")}
+	msg := &Message{Key: []byte("key"), Value: []byte("value"), Time: tm.Unix(0, 0).UTC()}
 
 	brokerWriterMock.EXPECT().WriteMessages(gomock.Any(), *msg).Return(errors.New("error"))
```

Comment:
here we are testing retruning error, you can pass gomock.Any() as `*msg`. We are testing msg in Success test.